openedx/core/djangolib/model_mixins.py

```diff
@@ -32,5 +32,10 @@ def delete_by_user_value(cls, value, field):
         Returns False otherwise.
         """
         filter_kwargs = {field: value}
-        num_deleted_records, _ = cls.objects.filter(**filter_kwargs).delete()
-        return num_deleted_records > 0
+        records_matching_user_value = cls.objects.filter(**filter_kwargs)
+
+        if not records_matching_user_value.exist():
```

Comment:
#NIT instead of doing negative here, it will be clearer to just do
```
if records_to_delete.exists():
    records_to_delete.delete()
    return True
```

Reply:
Just to be clear, you don't even need a `return False` at the end of the function. Because all python function implicitly do `return None`

Reply:
I agree that it's a little clearer to test the positive condition first, as in your code snippet.  I think it's actually nicer to explicitly return `False` at the end, so that this function will always return a boolean value.  A function that returns either `True` or `None` is a code smell, IMHO.

Reply:
And actually, I don't feel that strongly on the first point, either.  Doing an early return for the negative case is a totally valid stylistic choice (also IMHO).

Reply:
I don't have strong opinion on my suggestion either, that's why it's a NIT